pkg/lvmdriver/services/grpc_server.go

```diff
@@ -0,0 +1,103 @@
+package services
+
+import (
+	"net"
+	"os"
+	"sync"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"github.com/openshift/lvm-driver/pkg/lvmdriver/utils"
+	"google.golang.org/grpc"
+	"k8s.io/klog/v2"
+)
+
+type GrpcServer interface {
+	// start the service
+	Start()
+	// waits for the service to stop
+	Wait()
+	// Graceful shutdown
+	Stop()
+	// Forced shutdown
+	ForceStop()
+}
+
+type GrpcServerConfig struct {
+	Endpoint   string
+	IdServer   csi.IdentityServer
+	NodeServer csi.NodeServer
+}
+
+// GrpcServer is the primary server for all k8s related communications
+type grpcServer struct {
+	wg         sync.WaitGroup
+	server     *grpc.Server
+	endpoint   string
+	idServer   csi.IdentityServer
+	nodeServer csi.NodeServer
+}
+
+func NewGrpcServer(config GrpcServerConfig) GrpcServer {
+	return &grpcServer{
+		endpoint:   config.Endpoint,
+		idServer:   config.IdServer,
+		nodeServer: config.NodeServer,
+	}
+}
+
+func (s *grpcServer) Start() {
```

Comment:
Why have separate Start / Wait when you anyhow need to start and wait directly?

Reply:
This was mostly a copy paste, will merge the two functions